File: final frontend mantrasetu/MantraSetu-Saarthi-main/src/services/kundali.service.ts
import apiClient from '@/api/api';

export interface KundaliGeneratePayload {
  name: string;
  dob: string;
  tob?: string;
  pob: string;
  gender?: string;
}

export interface KundaliHouseItem {
  house: number;
  sign?: string;
  planets?: string[];
  description?: string;
}

export interface KundaliGenerateResponse {
  id?: string;
  name?: string;
  chart_title?: string;
  houses?: KundaliHouseItem[];
  planetary_positions?: Record<string, string>;
  dasha_info?: string;
  insights?: string[];
  note?: string;
  message?: string;
}

export const kundaliService = {
  /**
   * POST /kundali/generate
   */
  async generateKundali(payload: KundaliGeneratePayload): Promise<KundaliGenerateResponse> {
    const response = await apiClient.post<KundaliGenerateResponse>('/kundali/generate', payload);
    return response.data;
  },

  /**
   * GET /kundali/history (Stub reserved for future development)
   */
  async getKundaliHistory(): Promise<unknown> {
    const response = await apiClient.get('/kundali/history');
    return response.data;
  },
};

export default kundaliService;

File: final backend mantrasetu/mantrasetu-saarthi-backend-main/app/services/kundali_service.py
from app.database.kundali_db import (
    create_kundali,
    get_kundali_history_by_user,
)

from app.schemas.kundali_schema import (
    GenerateKundaliRequest,
    GenerateKundaliResponse,
    KundaliHistoryItem,
)


class KundaliService:

    @staticmethod
    async def generate_kundali(
        user_id: str,
        data: GenerateKundaliRequest,
    ) -> GenerateKundaliResponse:

        kundali_id = await create_kundali(
            user_id=user_id,
            name=data.name,
            dob=data.dob,
            tob=data.tob,
            pob=data.pob,
            gender=data.gender,
        )

        return GenerateKundaliResponse(
            status="success",
            message="Kundali generated successfully.",
            kundali_id=kundali_id,
        )

    @staticmethod
    async def get_kundali_history(
        user_id: str,
    ) -> list[KundaliHistoryItem]:

        history = await get_kundali_history_by_user(user_id)

        return [
            KundaliHistoryItem(**item)
            for item in history
        ]